src/Microsoft.AspNetCore.Mvc.ViewFeatures/Internal/ExpressionTextCache.cs

```diff
@@ -0,0 +1,86 @@
+                new ConcurrentDictionary<LambdaExpression, string>(LambdaExpressionComparer.Instance);
+
+        private class LambdaExpressionComparer : IEqualityComparer<LambdaExpression>
+        {
+            public static readonly LambdaExpressionComparer Instance = new LambdaExpressionComparer();
+
+            public bool Equals(LambdaExpression lambdaExpression1, LambdaExpression lambdaExpression2)
+            {
+                // We will cache only pure member access expressions. Hence we compare two expressions 
+                // to be equal only if they are identical member access expressions.
+                var expression1 = lambdaExpression1.Body;
+                var expression2 = lambdaExpression2.Body;
+
+                while (expression1 != null && expression2 != null)
+                {
+                    if (!expression1.EqualsTo(expression2))
+                    {
+                        return false;
+                    }
+
+                    if (expression1.NodeType == ExpressionType.MemberAccess)
+                    {
+                        var memberExpression1 = (MemberExpression)expression1;
+                        expression1 = memberExpression1.Expression;
+
+                        var memberExpression2 = (MemberExpression)expression2;
+                        expression2 = memberExpression2.Expression;
+                    }
+                    else if (expression1.NodeType == ExpressionType.Parameter)
+                    {
+                        expression1 = expression2 = null;
+                    }
+                    else
+                    {
+                        // should never come here.
+                        return false;
+                    }
+                }
+
+                return true;
+            }
+
+            public int GetHashCode(LambdaExpression lambdaExpression)
+            {
+                var expression = lambdaExpression.Body;
+                var hashCode = 0;
+
+                while (expression != null)
+                {
+                    if (expression.NodeType == ExpressionType.MemberAccess)
+                    {
+                        var memberExpression = (MemberExpression)expression;
+                        hashCode ^= memberExpression.Member.Name.GetHashCode();
+                        expression = memberExpression.Expression;
+                    }
+                    else if (expression.NodeType == ExpressionType.Parameter)
+                    {
+                        var parameterExpression = (ParameterExpression)expression;
+                        hashCode ^= parameterExpression.Name.GetHashCode();
+                        expression = null;
+                    }
+                    else
+                    {
+                        return lambdaExpression.GetHashCode();
+                    }
+                }
+
+                return hashCode;
+            }
```

Comment:
Rewrite this method to use `HashCodeCombiner`. You'll see a number of examples in this repo.